flytekit/types/file/file.py

```diff
@@ -26,6 +26,13 @@ def noop():
 T = typing.TypeVar("T")
 
 
+def _get_origin_type(t: Type) -> Type:
+    """Return the origin type for annotated types or the type itself, such that it can be used with ``issubclass()``"""
+    if get_origin(t) is Annotated:
+        return get_args(t)[0]
```

Comment:
we do this all the place in the `type_engine.py`. https://github.com/flyteorg/flytekit/blob/a1636a89daee2301a056b3eba41fe4dcea1257b7/flytekit/core/type_engine.py#L737-L748
could you move this function to `type_engine.py`, and import it instead.

Reply:
Moved the function there and refactored `type_engine.py` quite a bit in 093536fef. This should also make the handling of annotated types there more uniform (and added a few tests).

Is this what you had in mind?